### PR TITLE
Use global frame for transformLaserScanToPointCloud.

### PR DESCRIPTION
**Pull request by ZdenekM**

### CODE REVIEW

File: costmap_2d/plugins/obstacle_layer.cpp [hunk 1]
```diff
@@ -190,10 +190,12 @@ void ObstacleLayer::laserScanCallback(const sensor_msgs::LaserScanConstPtr& mess
   sensor_msgs::PointCloud2 cloud;
   cloud.header = message->header;
 
+  if (!tf_->waitForTransform(global_frame_, message->header.frame_id, message->header.stamp + ros::Duration(message->scan_time), ros::Duration(0.25))) return;
+
```
Comment (DLu) [inline costmap_2d/plugins/obstacle_layer.cpp:199]: Please add a comment for what this does and break it up into multiple lines.

Comment (ZdenekM) [inline costmap_2d/plugins/obstacle_layer.cpp:199]: Sure. By the way - is it ok to keep there some fixed timeout?

Comment (DLu) [inline costmap_2d/plugins/obstacle_layer.cpp:199]: It'd be better to use a constant, although there isn't a transform_tolerance variable anywhere handy. What use case does this fix again?

Comment (ZdenekM) [inline costmap_2d/plugins/obstacle_layer.cpp:199]: See this: https://github.com/ros-perception/laser_geometry/blob/hydro-devel/include/laser_geometry/laser_geometry.h#L156.

Comment (DLu) [inline costmap_2d/plugins/obstacle_layer.cpp:199]: That explains the new reference frame. Does it require the waitForTransform as well?

Comment (ZdenekM) [inline costmap_2d/plugins/obstacle_layer.cpp:199]: Well, not exactly :) But if you check also source code (https://github.com/ros-perception/laser_geometry/blob/hydro-devel/src/laser_geometry.cpp#L217), there is no waitForTransform. Without it I'm getting a lot of exceptions.

Comment (ZdenekM) [inline costmap_2d/plugins/obstacle_layer.cpp:199]: Just found that necessity of waiting for transform is documented here: http://wiki.ros.org/laser_geometry#High_fidelity_projection.

Comment (DLu) [inline costmap_2d/plugins/obstacle_layer.cpp:199]: Ok. Looks good. Just break it up into two+ lines and I'll merge.

**Comment by ZdenekM** [comment]
Is it fine like this? I also added some debug info - otherwise it might be hard to know what's going on.

**Comment by DLu** [comment]
Looking good. Maybe this is just being pedantic, but can you add a newline after "message->header.frame_id," just so that line isn't so long?

**Comment by ZdenekM** [comment]
Fixed. Proper formatting is not my strength :) Thanks for notes regarding this.

**Comment by DLu** [comment]
Thanks for the contributions!

**Pull request merged**